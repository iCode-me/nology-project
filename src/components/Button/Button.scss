@use "../../styles/mixins" as *;
@use "../../styles/variables" as *;

.button {
  @include focus-outline;

  border: none;
  cursor: pointer;
  background-color: $primary-color;
  border-radius: 0.625rem;
  padding: 0.5rem 2rem;
  width: 100%;
  transition: all 0.1s ease-in-out;

  &:disabled {
    cursor: not-allowed;
    background-color: $disabled-background-color;
    color: $disabled-text-color;
  }

  &:hover {
    background-color: darken($primary-color, 5%);
  }
}

// at tablet breakpoints, button is no longer full width
@media screen and (min-width: $tablet-breakpoint) {
  .button {
    width: unset;
  }
}
