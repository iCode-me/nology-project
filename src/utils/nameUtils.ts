export const formatName = (firstName: string, lastName: string): string => {
  return `${firstName} ${lastName}`;
};

// export const formatNameArray = (names: string[]): string => {
//   return names.join(" ");
// };
