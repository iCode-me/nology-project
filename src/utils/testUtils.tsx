import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

type CustomRenderOptions = {
  useRouting: boolean;
};

export const customRender = (
  ui: JSX.Element,
  options: CustomRenderOptions = {
    useRouting: true,
  }
) => {
  let uiResult = ui;

  if (options.useRouting) {
    uiResult = wrapWithRouting(ui);
  }

  return render(uiResult);
};

const wrapWithRouting = (ui: JSX.Element): JSX.Element => {
  return <BrowserRouter>{ui}</BrowserRouter>;
};
