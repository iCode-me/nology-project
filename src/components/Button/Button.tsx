import classname from "classnames";
import "./Button.scss";

type ButtonProps = {
  className?: string;
  disabled?: boolean;
  label: string;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  type?: "submit" | "button" | "reset";
};

const ROOT_CLASSNAME = "button";

const Button: React.FC<ButtonProps> = ({
  className,
  disabled,
  label,
  onClick,
  type = "button",
}) => {
  //classname allows us to combine multiple classes
  const rootClasses = classname(className, ROOT_CLASSNAME);

  return (
    <button
      className={rootClasses}
      disabled={disabled}
      onClick={onClick}
      type={type}
    >
      {label}
    </button>
  );
};

export default Button;
