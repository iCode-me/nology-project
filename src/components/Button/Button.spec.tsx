import { fireEvent, screen } from "@testing-library/react";
import { customRender } from "../../utils/testUtils";
import Button from "./Button";

export default describe("Button", () => {
  const mockOnClick = jest.fn();
  const labelText = "Button Label";

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("Renders", () => {
    const { container } = customRender(<Button label={labelText} />);

    expect(container).toMatchSnapshot();
  });

  it("Doesn't call onClick when disabled", () => {
    customRender(<Button onClick={mockOnClick} label={labelText} disabled />);

    const button = screen.getByRole("button");
    fireEvent.click(button);

    expect(mockOnClick).not.toHaveBeenCalled();
  });

  it("Calls onClick when enabled", () => {
    customRender(<Button onClick={mockOnClick} label={labelText} />);

    const button = screen.getByRole("button");
    fireEvent.click(button);

    expect(mockOnClick).toBeCalledTimes(1);
  });
});
