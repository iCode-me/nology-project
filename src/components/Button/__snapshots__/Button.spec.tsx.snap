// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders 1`] = `
<div>
  <button
    class="button"
    type="button"
  >
    Button Label
  </button>
</div>
`;
